lib/student_greeter.rb

```diff
@@ -0,0 +1,84 @@
+class StudentGreeter
+  def self.greet_all_ungreeted_students
+    new.greet_all_ungreeted_students
+  end
+
+  def initialize
+    @greeters = User.where(greeter: true)
+  end
+
+  def greet_all_ungreeted_students
+    courses = Course.strictly_current
+    courses.each do |course|
+      nonstudents_in_course = course.nonstudents
+      possible_greeters = @greeters & nonstudents_in_course
```

Comment:
possible greeters for a course are users who are both a) enrolled as nonstudents, and b) have the greeter flag.